integration/test_timescaledb.py

```diff
@@ -230,48 +215,45 @@ def test_timescaledb_all_types_and_precision(timescaledb):
         assert f"varchar_test_{i}" in row[8]  # varchar contains expected pattern
 
     # Test precision-specific functionality within the same test
-    cursor = connection.cursor()
-    if timestamp_precision == "milliseconds":
-        # Test millisecond precision
-        test_time = "2019-09-18 01:55:10.123+00"
-        cursor.execute(
-            f"""
-            INSERT INTO {table_name} (time, binary_col, bool_col, double_col, float_col,
-                                    int_col, timestamp_col, nchar_col, varchar_col)
-            VALUES (%s, %s, %s, %s, %s, %s, %s, %s, %s)
-        """,
-            (test_time, b"test", True, 1.23, 4.56, 999, test_time, "MS", "millisecond_test"),
-        )
-
-        cursor.execute(f"SELECT EXTRACT(MILLISECONDS FROM time) FROM {table_name} WHERE int_col = 999")
-        result = cursor.fetchone()
-        assert result[0] >= 10000  # At least 10 seconds worth of milliseconds
-
-    elif timestamp_precision == "microseconds":
-        # Test microsecond precision
-        test_time = "2019-09-18 01:55:10.123456+00"
-        cursor.execute(
-            f"""
-            INSERT INTO {table_name} (time, binary_col, bool_col, double_col, float_col,
-                                    int_col, timestamp_col, nchar_col, varchar_col)
-            VALUES (%s, %s, %s, %s, %s, %s, %s, %s, %s)
-        """,
-            (test_time, b"test", True, 1.23, 4.56, 999, test_time, "US", "microsecond_test"),
-        )
-
-        cursor.execute(f"SELECT EXTRACT(MICROSECONDS FROM time) FROM {table_name} WHERE int_col = 999")
-        result = cursor.fetchone()
-        assert result[0] >= 10000000  # At least 10 seconds worth of microseconds
-
-    cursor.close()
+    with psycopg.connect(dsn_url, autocommit=True) as connection:
+        with connection.cursor() as cursor:
+            if timestamp_precision == "milliseconds":
+                # Test millisecond precision
+                test_time = "2019-09-18 01:55:10.123+00"
+                cursor.execute(
+                    f"""
+                    INSERT INTO {table_name} (time, binary_col, bool_col, double_col, float_col,
+                                            int_col, timestamp_col, nchar_col, varchar_col)
+                    VALUES (%s, %s, %s, %s, %s, %s, %s, %s, %s)
+                """,
+                    (test_time, b"test", True, 1.23, 4.56, 999, test_time, "MS", "millisecond_test"),
+                )
+
+                cursor.execute(f"SELECT EXTRACT(MILLISECONDS FROM time) FROM {table_name} WHERE int_col = 999")
+                result = cursor.fetchone()
+                assert result[0] >= 10000  # At least 10 seconds worth of milliseconds
```

Comment:
This helps readability a bit
```suggestion
                assert result[0] >= 10_000  # At least 10 seconds worth of milliseconds
```

Reply:
will fix